frontend/components/Main/MainHeader/index.tsx

```diff
@@ -274,27 +277,17 @@ export const MainHeader = () => {
         totalEthBalance > requiredGas
       );
     })();
-
     const serviceExists = !!services?.[0];
 
-    if (!isDeployable) {
-      return (
-        <Button type="default" size="large" disabled>
-          Start agent {!serviceExists && '& stake'}
-        </Button>
-      );
-    }
+    const buttonProps: ButtonProps = {
+      type: 'primary',
+      size: 'large',
+      disabled: !isEligibleForStakingAction || !isDeployable,
+      onClick: canStartEvictedAgent || isDeployable ? handleStart : undefined,
```

Comment:
I wonder, would it improve readability if we move `canStartEvictedAgent` into `isDeployable`?